### DISCUSSION TITLE
Turbine power with turbine weights

### DISCUSSION THREAD

**Discussion by Teddybear0227** [NatLabRockies/floris#896] [Q&A]
I am currently working on a project involving offshore wind farms. I am trying to find the inter-farm wake effect between 3 wind farms and need turbine powers for a single wind farm. I tried using fi.get_turbine_powers with turbine weights to only get the power data for the specified wind farm while considering wake effects of all three wind farms but it does not seem to be working. Is there any alternative to find the turbine powers of a single wind farm with wake effects of 3 wind farms in cvonsideration?

**Comment by misi9170** [comment]
Hi @Teddybear0227 , you can simply index the output of `fi.get_turbine_powers()` with the indices of the turbines in your farm of interest. Here is an example (based on FLORIS v4---for FLORIS v3, you will need an extra `:` when indexing the output of `get_turbine_powers()`):

```python
import numpy as np

from floris import FlorisModel

fmodel = FlorisModel("inputs/gch.yaml")

# Let's assume the first two positions are from a farm you want the powers from, 
# and the second two are from a farm you do not want the powers from
fmodel.set(layout_x=[0, 500.0, 1000.0, 1500.0], layout_y=[0.0, 0.0, 0.0, 0.0])
turbine_indices_of_interest = [0, 1]

fmodel.set(
    wind_directions=np.array([270.0, 270.0, 270.0, 270.0]),
    wind_speeds=[8.0, 8.0, 10.0, 10.0],
    turbulence_intensities=np.array([0.06, 0.06, 0.06, 0.06]),
)

fmodel.run()

# Now, we can use the indices of the farm we want to extract the turbine powers of interest
turbine_powers = fmodel.get_turbine_powers()[:, turbine_indices_of_interest] / 1000.0

print("The turbine power matrix should be of dimensions 4 (n_findex) X 2 (number of turbines in farm of interest)")
print(turbine_powers)
print("Shape: ", turbine_powers.shape)
```

**Reply by paulf81** [reply]
There is also an example using turbine_weights: https://nrel.github.io/floris/examples/009_compare_farm_power_with_neighbor.html
